src/runtime/hexagon_dma_device_interface.cpp

```diff
@@ -0,0 +1,446 @@
+/**
+*  This file contains the definitions of the hexagon dma device interface class
+* This file is an interface between Halide Runtime and the DMA driver
+* The functions in this file take care of the various functionalities of the
+* DMA driver through their respective halide device interface functions
+* The function in this file use the Hexagon DMA context structure for sharing
+* the DMA context across various APIs of the
+* DMA device interface
+*/
+
+#include "runtime_internal.h"
+#include "device_buffer_utils.h"
+#include "device_interface.h"
+#include "printer.h"
+#include "mini_qurt.h"
+#include "hexagon_mini_dma.h"
+#include "hexagon_dma_device_shim.h"
+#include "HalideRuntimeHexagonDma.h"
+#include "hexagon_dma_rt.h"
+#include "hexagon_dma_context.h"
+
+using namespace Halide::Runtime::Internal;
+using namespace Halide::Runtime::Internal::Qurt;
+
+extern "C" {
+
+extern WEAK halide_device_interface_t hexagon_dma_device_interface;
+
+int halide_dma_device_release(void *user_context) {
```

Comment:
These routines need to be named "halide_hexagon_dma_" unless this is a general dma device.

Reply:
They also need to be declared WEAK.